framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -1839,20 +1859,28 @@ public void postAsMemberOf(
      */
```

Comment:
Could you update the javadoc?
This now also applies the `defaultForTypeAnnotator`, which is surprising given the method name.
It looks like `null` is passed as element to change what would happen by default, so calling that out would be good.
I was also surprised by the special case for primitives, so giving a reason for that would be good.